bundles/org.eclipse.jface.text/src/org/eclipse/jface/text/contentassist/IContentAssistProcessorExtension2.java

```diff
@@ -0,0 +1,113 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Dawid Pakuła and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Dawid Pauła - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jface.text.contentassist;
+
+import org.eclipse.jface.text.ITextViewer;
+
+/**
+ * Extension interface of {@link IContentAssistProcessor} that forward information about
+ * autoActivation
+ *
+ * @since 3.24
+ */
+public interface IContentAssistProcessorExtension2 extends IContentAssistProcessor {
```

Comment:
Why not default methods?

Reply:
This was my original idea but in this case IDE will not suggest you to implement them immediately

Reply:
I don't think this argument is very strong: if one is to implement a ContentAssistProcessor, the IDE wouldn't suggest them to implement IContentAssistProcessExtension2 and its method either. So to me, the effort to discover and adopt those new methods is very comparable, and even smaller IMO with default method if they're invoked by the CotnentAssistant (to transitively delegate to older variants) and part of the call stack that we see when debugging. We could even consider deprecating the older methods with a note that this code isn't invoked directly any more byt the ContentAssistant and to recommend implementing the newer variants, and then add some default empty implementations.

Reply:
I prepared defaults and dropped IContentAssistProcessorExtension